app/core/core-design-system/src/main/kotlin/com/hedvig/android/core/designsystem/component/information/HedvigInformationSection.kt

```diff
@@ -69,16 +67,15 @@ fun HedvigInformationSection(
     )
     Spacer(Modifier.height(2.dp))
     if (subTitle != null) {
-      CompositionLocalProvider(LocalContentColor provides MaterialTheme.colorScheme.onSurfaceVariant) {
```

Comment:
That was something I wanted to ask you about. Is it womething that we don't want it to be used everywhere, only on top of the Composable tree?

Reply:
Hmm, it's a bit of a nuanced discussion. And how interesting timing where I just made this PR as well https://github.com/HedvigInsurance/android/pull/2063
The short version is that when we can use things explicitly it's just better to do it this way instead of using a composition local. Especially here where the usage is literally just a couple of lines further down, it was a mistake from me to do it this way.
It only makes sense to provide such composition locals in scenarios like this one https://github.com/HedvigInsurance/android/blob/71f939aaf99c645e5a7c67ae980b6aca0717eaf1/app/core/core-design-system/src/main/kotlin/com/hedvig/android/core/designsystem/component/textfield/HedvigTextFieldImpl.kt#L235-L241 where we do not actually know what the composable content has inside of it, and we can not pass the color in there instead. So there we pass it as a composition local so that the content can use it if they want. But that still is just that we provide it, the content itself may just decide not to use it.
More on locals here https://developer.android.com/jetpack/compose/compositionlocal
More on why they are often not the right choice here 

Some more thoughts on this from one of the compose engineers in Google:
https://slack-chats.kotlinlang.org/t/13866799/ambients-were-renamed-to-compositionlocal-keep-in-mind-that-#16dc0903-b0c6-437d-be88-9a1e060c4750

They got their place, the App's theme being one of them. But overall it's hard to justify using in normal code.